packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +869,777 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+  }
 }
 
-// A CustomPainter to draw the divider lines.
+// iOS style layout policy for sizing and positioning an alert dialog's action
+// buttons.
+//
+// The policy is as follows:
+//
+// If a single action button is provided, or if 2 action buttons are provided
+// that can fit side-by-side, then action buttons are sized and laid out in a
+// single horizontal row. The row is exactly as wide as the dialog, and the row
+// is as tall as the tallest action button. A horizontal divider is drawn above
+// the button row. If 2 action buttons are provided, a vertical divider is
+// drawn between them. The thickness of the divider is set by [dividerThickness].
 //
-// Draws the cross-axis divider lines, used when the layout is vertical.
-class _CupertinoVerticalDividerPainter extends CustomPainter {
-  _CupertinoVerticalDividerPainter();
+// If 2 action buttons are provided but they cannot fit side-by-side, then the
+// 2 buttons are stacked vertically. A horizontal divider is drawn above each
+// button. The thickness of the divider is set by [dividerThickness]. The minimum
+// height of this [RenderBox] in the case of 2 stacked buttons is as tall as
+// the 2 buttons stacked. This is different than the 3+ button case where the
+// minimum height is only 1.5 buttons tall. See the 3+ button explanation for
+// more info.
+//
+// If 3+ action buttons are provided then they are all stacked vertically. A
+// horizontal divider is drawn above each button. The thickness of the divider
+// is set by [dividerThickness]. The minimum height of this [RenderBox] in the case
+// of 3+ stacked buttons is as tall as the 1st button + 50% the height of the
+// 2nd button. In other words, the minimum height is 1.5 buttons tall. This
+// minimum height of 1.5 buttons is expected to work in tandem with a surrounding
+// [ScrollView] to match the iOS dialog behavior.
+//
+// Each button is expected to have an _ActionButtonParentData which reports
+// whether or not that button is currently pressed. If a button is pressed,
+// then the dividers above and below that pressed button are not drawn - instead
+// they are filled with the standard white dialog background color. The one
+// exception is the very 1st divider which is always rendered. This policy comes
+// from observation of native iOS dialogs.
+class _RenderCupertinoDialogActions extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, MultiChildLayoutParentData>,
+        RenderBoxContainerDefaultsMixin<RenderBox, MultiChildLayoutParentData> {
+  _RenderCupertinoDialogActions({
+    List<RenderBox> children,
+    @required double dialogWidth,
+    double dividerThickness = 0.0,
+  }) : _dialogWidth = dialogWidth,
+       _dividerThickness = dividerThickness {
+    addAll(children);
+  }
+
+  double get dialogWidth => _dialogWidth;
+  double _dialogWidth;
+  set dialogWidth(double newWidth) {
+    if (newWidth != _dialogWidth) {
+      _dialogWidth = newWidth;
```

Comment:
nit: be consistent with if != and if == below

Reply:
Fixed.